src/Test/L0/HostContextL0.cs

```diff
@@ -53,71 +53,143 @@ public void GetServiceReturnsSingleton()
             }
         }
 
+        private static readonly (string, string)[] _urlSecretCases = new[]
+        {
+            // some URLs with secrets to mask
+            ("https://user:pass@example.com/path", "https://user:***@example.com/path"),
+            ("http://user:pass@example.com/path", "http://user:***@example.com/path"),
+            ("ftp://user:pass@example.com/path", "ftp://user:***@example.com/path"),
+            ("https://user:pass@example.com/weird:thing@path", "https://user:***@example.com/weird:thing@path"),
+            ("https://user:pass@example.com:8080/path", "https://user:***@example.com:8080/path"),
+            ("https://user:pass@example.com:8080/path\nhttps://user2:pass2@example.com:8080/path", "https://user:***@example.com:8080/path\nhttps://user2:***@example.com:8080/path"),
+            ("https://user@example.com:8080/path\nhttps://user2:pass2@example.com:8080/path", "https://user@example.com:8080/path\nhttps://user2:***@example.com:8080/path"),
+            ("https://user:pass@example.com:8080/path\nhttps://user2@example.com:8080/path", "https://user:***@example.com:8080/path\nhttps://user2@example.com:8080/path"),
+            // some URLs without secrets to mask
+            ("https://example.com/path", null),
+            ("http://example.com/path", null),
+            ("ftp://example.com/path", null),
+            ("ssh://example.com/path", null),
+            ("https://example.com/@path", null),
+            ("https://example.com/weird:thing@path", null),
+            ("https://example.com:8080/path", null),
+        };
+
+        public static readonly SecretCases UrlSecrets_NewMasker = new(_urlSecretCases, useNewSecretMasker: true);
+        public static readonly SecretCases UrlSecrets_LegacyMasker = new(_urlSecretCases, useNewSecretMasker: false);
```

Comment:
The additional masking regexes flag has no impact on url secret masking, which, as before is always done. Note to self, though, it would be prudent to add the four cases here to test that this is so.

Reply:
Related: there is no coverage that other things that are always done irrespective of flags, are indeed always done: JSON/Uri/Backslash escape handling. This is pre-existing but given the refactoring in HostContext of SecretMasker initialization, it would be good to add this now. Again, note to self.